include/event_dbus_monitor.hpp

```diff
@@ -0,0 +1,640 @@
+#include <dbus_utility.hpp>
+#include <error_messages.hpp>
+#include <event_service_manager.hpp>
+#include <resource_messages.hpp>
+
+#include <memory>
+#include <string>
+#include <variant>
+
+namespace crow
+{
+namespace dbus_monitor
+{
+
+static std::shared_ptr<sdbusplus::bus::match::match> matchHostStateChange;
+static std::shared_ptr<sdbusplus::bus::match::match> matchBMCStateChange;
+static std::shared_ptr<sdbusplus::bus::match::match>
+    matchVMIIPEnabledPropChange;
+static std::shared_ptr<sdbusplus::bus::match::match> matchVMIIPChange;
+static std::shared_ptr<sdbusplus::bus::match::match> matchVMIEthIntfPropChange;
+static std::shared_ptr<sdbusplus::bus::match::match> matchDumpCreatedSignal;
+static std::shared_ptr<sdbusplus::bus::match::match> matchDumpDeletedSignal;
+static std::shared_ptr<sdbusplus::bus::match::match> matchBIOSAttrUpdate;
+static std::shared_ptr<sdbusplus::bus::match::match> matchBootProgressChange;
+static std::shared_ptr<sdbusplus::bus::match::match> matchEventLogCreated;
+static std::shared_ptr<sdbusplus::bus::match::match> matchPostCodeChange;
+static std::shared_ptr<sdbusplus::bus::match::match> matchPlatformSAIChange;
+static std::shared_ptr<sdbusplus::bus::match::match> matchPartitionSAIChange;
+
+static uint64_t postCodeCounter = 0;
+
+void registerHostStateChangeSignal();
+void registerBMCStateChangeSignal();
+void registerVMIIPEnabledPropChangeSignal();
+void registerVMIIPChangeSignal();
+void registerVMIEthIntfPropChangeSignal();
+void registerDumpCreatedSignal();
+void registerDumpDeletedSignal();
+void registerBIOSAttrUpdateSignal();
+void registerBootProgressChangeSignal();
+void registerEventLogCreatedSignal();
+void registerPostCodeChangeSignal();
+void registerSAIStateChangeSignal();
+
+inline void sendEventOnEthIntf(const std::string& origin)
+{
+    redfish::EventServiceManager::getInstance().sendEvent(
+        redfish::messages::resourceChanged(), origin, "EthernetInterface");
+}
+
+inline void vmiNwPropertyChange(sdbusplus::message::message& msg)
+{
+    if (msg.is_method_error())
+    {
+        BMCWEB_LOG_ERROR(
+            "BMC Hypervisor Network properties changed Signal error");
+        return;
+    }
+
+    std::string objPath = msg.get_path();
+
+    if (!(objPath.starts_with(
+            "/xyz/openbmc_project/network/hypervisor/eth0")) &&
+        !(objPath.starts_with("/xyz/openbmc_project/network/hypervisor/eth1")))
+    {
+        return;
+    }
+
+    std::string infName;
+    if (objPath.find("/eth0") != std::string::npos)
+    {
+        infName = "eth0";
+    }
+    else if (objPath.find("/eth1") != std::string::npos)
+    {
+        infName = "eth1";
+    }
+    else
+    {
+        BMCWEB_LOG_ERROR("Unsupported Interface name");
+        return;
+    }
+
+    std::string origin = "/redfish/v1/Systems/hypervisor/EthernetInterfaces/" +
+                         infName;
```

Comment:
I think the suggested style is for more clarity.
```
    std::string origin = std::format("/redfish/v1/Systems/hypervisor/EthernetInterfaces/{}",  infName);
```

Reply:
> I think the suggested style is for more clarity.
> 
> ```
>     std::string origin = std::format("/redfish/v1/Systems/hypervisor/EthernetInterfaces/{}",  infName);
> ```

done